internal/console/console.go

```diff
@@ -341,20 +341,34 @@ func getHeader(columns []string) string {
 			header += fmt.Sprintf("%-40s", column)
 		case "Port":
 			header += fmt.Sprintf("%-5s", column)
+		case "Status":
+			header += fmt.Sprintf("%-11s", column)
+		case "Speed":
+			header += fmt.Sprintf("%-11s", column)
+		case "ETA":
+			header += fmt.Sprintf("%-8s", column)
+		case "Size":
+			header += fmt.Sprintf("%-9s", column)
 		default:
-			panic(fmt.Sprintf("unsupported column %s", column))
+			// Add column without formatting
+			header += column
 		}
 	}
 	return header
 }
 
-func getRow(columns []string, t rpctypes.Torrent, index int) string {
+func getRow(c *Console, t rpctypes.Torrent, index int) string {
 	row := ""
-	for i, column := range columns {
+	for i, column := range c.columns {
 		if i != 0 {
 			row += " "
 		}
 
+		stats, err := c.client.GetTorrentStats(t.ID)
```

Comment:
I don't think calling `GetTorrentStats` method for every torrent is a good idea. This will make:
* rendering torrent list slower
* puts unnecessary load on the server
* and has a risk of a stuck `Torrent` loop may block the UI indefinitely.

Please not that Rain is designed for handling hundreds of torrents simultaneously.

Maybe we can have system for pushing changes from torrents without the need to query them one by one. Then, the UI can subscribe to those changes and show the necessary information to the user.

What do you think?

Reply:
Having a system that will push changes when they occur will cause the same issue if you seed hundreds of torrents simultaneously. Think best solution is to grab the torrent stats like this, but only do it for the torrents currently displayed on the screen. I have updated my pull request with this behavior.

It now shows empty rows when they are outside the view (without calling GetTorrentStats). It should probably avoid looping over torrents outside the view completely, but that would require modification to how current selected row works. Displaying an empty row works for now.